src/Main/MainGui.cpp

```diff
@@ -95,6 +95,35 @@ class Redirection
     FILE* file;
 };
 
+static void DisplayInfo(const QString& msg, bool preformatted = true)
+{
+    if (App::Application::Config()["Console"] == "1") {
+        std::cout << msg.toLatin1().constData();
+        return;
+    }
+
+    QString appName = QString::fromLatin1(App::Application::Config()["ExeName"].c_str());
+    QMessageBox msgBox;
+    msgBox.setIcon(QMessageBox::Information);
+    msgBox.setWindowTitle(appName);
+    msgBox.setDetailedText(msg);
+    msgBox.setText(preformatted ? QLatin1String("<pre>") + msg + QLatin1String("</pre>") : msg);
+    msgBox.exec();
+}
+
+static void DisplayCritical(const QString& msg, bool preformatted = true)
+{
+    if (App::Application::Config()["Console"] == "1") {
+        std::cerr << msg.toLatin1().constData();

```

Comment:
How about using std::string instead of going to the raw pointer access?
```c++
std::cerr << msg.toStdString();
```

Reply:
Much better indeed; same for the other suggestions. Thanks.